tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -79,6 +80,9 @@ def deploy_function_app(parameters):
             app_resource_group_name = ResourceIdParser.get_resource_group(app_id)
             app_service_plan = web_client.app_service_plans.get(app_resource_group_name, app_name)
 
+            # enable autoscale if applicable
```

Comment:
There is no need to enable autoscale if function_app already exists. We decided to avoid making any changes if user uses pre-provisioned app service plan.

Following this approach, I suggest to make autoscale enablement to be a part of `app_service_plan.deploy()`, that way we can avoid overriding existing configurations.

Also I think there needs to be a check because we can't enable autoscale for consumption functions (AppServicePlan SKU Dynamic). We can check it in the validation function